### DISCUSSION TITLE
Can I send raw packets via WEB UI or Telnet? Emulate EM10?

### DISCUSSION THREAD

**Discussion by Mike-FUT** [emsesp/EMS-ESP32#1165] [General]
Hi All,

I have a Buderus GB162 boiler which is connected to a EM10 module. The EM10 module is used for telling the boiler either the set value of the flow temperature (°C) or the set value of the burner power (%).

When I run "watch on" and connect 5 V to the EM10 this gives these values for the set flow temperature:
`[emsesp] alert(0x12) -W-> boiler(0x08), UBASetPoints(0x1A), data: 32`   (for setting it to 50 °C)
`[emsesp] alert(0x12) -W-> boiler(0x08), UBASetPoints(0x1A), data: 00`   (for setting it to off)

That looks exactly the same for the set burner power:
`[emsesp] alert(0x12) -W-> boiler(0x08), UBASetPoints(0x1A), data: 32`   (for setting it to 50 %)
`[emsesp] alert(0x12) -W-> boiler(0x08), UBASetPoints(0x1A), data: 00`   (for setting it to 0 %)

I thought there's no way that the boiler can understand if it is % or °C requested so I was checking the raw packets ("watch raw"):
`[telegram] Rx: 12 08 1A 00 00 11` (for setting the set flow temperature to "off")
`[telegram] Rx: 12 08 1A 01 00 13`  (for setting the set burner power to "0 %")

So there is a different offset (as expected) but this is only shown when displaying raw bytes.

Many questions for this topic:
- Since EM10 is such an important device and basically THE buderus standard for controlling the heating with a home automation system is it possible to add this as simple commands to set the "set flow temperature °C" or "set burner power %" like it would be done with connecting a 0-10 V signal to EM10 so I can get rid of this old fashioned 0-10 V link?
- Since right now it can't be directly sent is there a way for me to send raw packets?
- If i send raw bytes do I have to calculate CRC or will EMS-ESP do that for me and attach it?
- Is it possible to show these rather important values somewhere?

Sorry guys if any of these questions are stupid but I'm brand new to EMS-ESP and I couldn't find any of these answered and also I couldn't find anything related to EM10 which surprised me since to my knowledge that's the official way of controlling the boiler from external systems.

**Comment by MichaelDvP** [comment]
The log shows the offset in brackets if offset is a positive value. The 1A telegram in offset 0: flowtemp, offset1: burnpower,  
By writing to `selflowtemp` and `selburnpow` you can send the values to the boiler. 
For sending raw commands there is the command `system/send` with a string as parameter without CRC, e.g. in terminal `call system send "1B 08 1A 01 32"` for setting burnpower to 50%. (mqtt-topic system/send, API-URL: http://ems-esp.local/api/system/send)

Writing these values is temporary, the boier resets to default setting after ~3 minutes. So you have to repeat the commands every minute. That is what EM10 or thermostats do. In the 3.6 beta we have a scheduler where you can define such values. Create scheduler value with timer 00:01 and command `boiler/selburnpow`, value 0 and choose a name, this can also be selburnpower, but must be unique in scheduler. Than you can write a new value to  scheduler/selburnpower and this is sent to boiler every minute.

**Comment by Mike-FUT** [comment]
Amazing answer!
Thank you very much! When testing `selflowtemp` I selected a lower value than the current value so that means nothing will change when RC35 is sending a higher value at the same time.
But when using EM10 i select higher temperatures which then increases the "set flowtemp" so I thought `selflowtemp` doesn't work. I didn't recognize I was doing it all wrong...

So thank you again for bringing some light into the whole thing and how the EM10 works (sending every minute, I verified it already by watching "0x1A").

**Comment by horstvogel** [comment]
Hello, we have a Logano Plus gb312, which has an EM10 module, which is actually supposed to provide the external specification of the performance. I have to check if that ever worked. Now the EM10 module is recognized, but no values ​​are transferred via MQTT and I can also see it under the web interface. Are the values ​​exclusively set? In other words, you can't see anything in the web interface. Or  I have to add “customize”?

My goal is to check the EM10 is configured correctly... the installer says "everything is fine, I have no idea why it doesn't work..." 
the boiler always runs at full power and does not regulate at all.

Thanks


![image](https://github.com/user-attachments/assets/a83c35ba-8a0e-4a10-83f5-4c17a0c46b15)




![image](https://github.com/user-attachments/assets/563c7e89-3209-44bf-aee8-b277d274deb3)



![image](https://github.com/user-attachments/assets/2b0c856f-15cb-45d6-ab2d-272182074d23)

**Reply by proddy** [reply]
do you know what information you want to read?

All the data from the Alert module is included in your Boiler. See https://github.com/emsesp/EMS-ESP32/issues/575

**Comment by horstvogel** [comment]
But there is no _burner set power_ on the MC10 (boilder), 

_the boiler always runs at full power and does not regulate at all._  

I would like to see why the burner always runs at 100% and not only runs at 50% when set to 5 volts to the EM10 for example. 

That's why I thought you could see the external setpoint here

![image](https://github.com/user-attachments/assets/7a8c047f-2cb3-479a-b67e-6776e65b7736)

**Reply by horstvogel** [reply]
that was our problem. We removed the RC35 and the ems-esp and the 0-10 volt signal went through to the boiler again. It's just a pity that we can't use the ems-eps, presumably it also emulates an RC 35 or something? 




![image](https://github.com/user-attachments/assets/85e681e5-e706-490d-abf7-f0fb82e79fce)
